pkg/golinters/govet.go

```diff
@@ -59,20 +88,64 @@ func NewGovet(cfg *config.GovetSettings) *goanalysis.Linter {
 		unsafeptr.Analyzer,
 		unusedresult.Analyzer,
 	}
+}
+
+func isAnalyzerEnabled(name string, cfg *config.GovetSettings, defaultAnalyzers []*analysis.Analyzer) bool {
+	if cfg.EnableAll {
+		return true
+	}
+	// Raw for loops should be OK on small slice lengths.
+	for _, n := range cfg.Enable {
+		if n == name {
+			return true
+		}
+	}
+	for _, n := range cfg.Disable {
+		if n == name {
+			return true
+		}
+	}
+	if cfg.DisableAll {
+		return false
+	}
+	for _, a := range defaultAnalyzers {
+		if a.Name == name {
+			return true
+		}
+	}
+	return false
+}
+
+func analyzersFromConfig(cfg *config.GovetSettings) []*analysis.Analyzer {
```

Comment:
please, add minimal validation to `(r *FileReader) validateConfig()` by calling something like:
```go
func (cfg *config.GovetSettings) Validate() error {
    if cfg.EnableAll && cfg.DisableAll {
        return errors.New("enable-all and disable-all can't be combined")
    }
    if cfg.EnableAll && len(cfg.Enable) != 0) {
        return errors.New("enable-all and enable can't be combined")
    }
    if cfg.DisableAll && len(cfg.Disable) != 0) {
        return errors.New("disable-all and disable can't be combined")
    }
    return nil
}
```

Reply:
Done, thank you.
We probably want to validate that `cfg.Enable` contains valid analyzer names, but I'm not sure how to do it in that scope.